public/app/listeners/pipelineUtils.js

```diff
@@ -0,0 +1,313 @@
+    return (
+      !mainWindow.isDestroyed() &&
+      mainWindow?.webContents?.send(channel, {
+        type: EXECUTE_ALGORITHM_ERROR,
+        error: ERROR_EXECUTION_PROCESS,
+        log,
+      })
+    );
+  };
+};
+
+const runAlgorithm = (
+  mainWindow,
+  language,
+  algorithmFilepath,
+  filepath,
+  tmpPath,
+  parameters,
+  schemaId,
+  onRun,
+  onStop,
+  onSuccess,
+  onError,
+  clean,
+  channel,
+) => {
+  switch (language) {
+    case PROGRAMMING_LANGUAGES.PYTHON:
+      return executePythonAlgorithm(
+        { algorithmFilepath, filepath, tmpPath, parameters, schemaId },
+        { onRun, onStop, onSuccess, onError, clean },
+      );
+
+    default:
+      // check whether mainWindow still exist in case of
+      // the app quits before the process get killed
+      return (
+        !mainWindow.isDestroyed() &&
+        mainWindow.webContents.send(channel, {
+          type: EXECUTE_ALGORITHM_ERROR,
+          error: ERROR_UNKNOWN_PROGRAMMING_LANGUAGE,
+        })
+      );
+  }
+};
+
+const runPipelineAlgorithm = (
+  mainWindow,
+  db,
+  algorithms,
+  algorithm,
+  sourceId,
+  userProvidedFilename,
+  parameters,
+  schemaId,
+  indexAlgorithm,
+) => {
+  const { name: datasetName } = db
+    .get(DATASETS_COLLECTION)
+    .find({ id: sourceId })
+    .value();
+  const { name: algorithmName } = db
+    .get(ALGORITHMS_COLLECTION)
+    .find({ id: algorithm.id })
+    .value();
+  let execution;
+  if (indexAlgorithm + 1 === algorithms.length) {
+    execution = addExecutionObject(
+      db,
+      sourceId,
+      algorithm.id,
+      userProvidedFilename,
+      datasetName,
+      algorithmName,
+      db
+        .get(ALGORITHMS_COLLECTION)
+        .find(({ id }) => id === algorithm.id)
+        .value()?.parameters || [],
+      schemaId,
+    );
+  } else {
+    const newProvidedFileName = `${userProvidedFilename}alg:${algorithm.id}idx:${indexAlgorithm}`;
+    execution = addExecutionObject(
+      db,
+      sourceId,
+      algorithm.id,
+      newProvidedFileName,
+      datasetName,
+      algorithmName,
+      db
+        .get(ALGORITHMS_COLLECTION)
+        .find(({ id }) => id === algorithm.id)
+        .value()?.parameters || [],
+      schemaId,
+    );
+  }
+
+  const channel = buildExecuteAlgorithmChannel(execution.id);
+  const { filepath, description } = db
+    .get(DATASETS_COLLECTION)
+    .find({ id: sourceId })
+    .value();
+  const { filepath: algorithmFilepath, language } = db
+    .get(ALGORITHMS_COLLECTION)
+    .find({ id: algorithm.id })
+    .value();
+  const tmpPath = path.join(DATASETS_FOLDER, `tmp_${execution.id}.json`);
+
+  // prepare all callback functions
+  const onRun = getOnRunFunction(db, execution.id);
+  const onSuccess = ({ log }) => {
+    // save result in db
+    const newResult = createNewResultDataset(
+      {
+        name: execution.result.name?.length
+          ? execution.result.name
+          : `${datasetName}_${algorithmName}`,
```

Comment:
So should there be a `createNewResultDataset` for `executePipeline` and a `createNewResultDataset` for `executeAlgorithm`, because adding those modifications lead to two different `createNewResultDataset` for search type of execution.